pages/contributors.js

```diff
@@ -0,0 +1,220 @@
+import animationData from "../lottie/sad.json"
+import { Text } from "atomize"
+import { Spinner } from "react-bootstrap"
+import Lottie from "react-lottie"
+import Card from "../components/Cards/Card"
+
+const defaultOptions = {
+    loop: true,
+    autoplay: true,
+    animationData: animationData,
+}
+
+export default function Contributors() {
+    const [contriFrontend, setContributorsFront] = useState()
+    const [contriBackend, setContributorsBack] = useState()
+    const [contributors, setContributors] = useState()
+    const [error, setError] = useState([false, false, false])
+
+    useEffect(() => {
+        axios
+            .get(
+                "https://api.github.com/repos/COPS-IITBHU/hackalog-frontend/contributors"
+            )
+            .then((res) => setContributorsFront(res.data))
+            .catch((err) => {
+                let arr = error
+                arr[0] = true
+                setError(arr)
+            })
+        axios
+            .get(
+                "https://api.github.com/repos/COPS-IITBHU/hackalog-backend/contributors"
+            )
+            .then((res) => setContributorsBack(res.data))
+            .catch((err) => {
+                let arr = error
+                arr[1] = true
+                setError(arr)
+            })
+    }, [error])
+
+    if (!contributors) {
+        if (contriFrontend && contriBackend) {
+            let commonF = []
+            let commonB = []
+            let arr = []
+            for (let x of contriFrontend) {
+                for (let y of contriBackend) {
+                    if (x.login === y.login) commonF.push(x), commonB.push(y)
+                }
+            }
+
+            for (let x of commonF) {
+                arr.push({
+                    handle_name: x.login,
+                    name: x.login,
+                    handle_url: x.url,
+                    image: x.avatar_url,
+                    github: x.html_url,
+                    description: ["hackalog-frontend", "hackalog-backend"],
+                })
+            }
+            for (let x of contriFrontend) {
+                if (!commonF.includes(x))
+                    arr.push({
+                        handle_name: x.login,
+                        handle_url: x.url,
+                        name: x.login,
+                        image: x.avatar_url,
+                        github: x.html_url,
+                        description: ["hackalog-frontend"],
+                    })
+            }
+            for (let x of contriBackend) {
+                if (!commonB.includes(x))
+                    arr.push({
+                        handle_name: x.login,
+                        handle_url: x.url,
+                        name: x.login,
+                        image: x.avatar_url,
+                        github: x.html_url,
+                        description: ["hackalog-backend"],
+                    })
+            }
+            setContributors(arr)
+        }
```

Comment:
- Its not a good practise to add logic blocks directly inside functions, its better to use hooks such as `useEffect`.
- Add this block of code inside a `useEffect` hook with dependencies `contriFrontend` & `contriBackend`

Reply:
Yes, I am correcting it